File: Food.WebApi/Models/Restaurant.cs
using Microsoft.Extensions.Hosting;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Food.WebApi.Models
{
    public class Restaurant
    {
        public Restaurant()
        {

        }

        [Key]
        public int? Id { get; set; }
        public string Image { get; set; }
        public string Name { get; set; }
        [JsonPropertyName("ownerid")]
        public int OwnerId { get; set; }

        public ICollection<Dish> Dishes { get; set; } = new List<Dish>();
        public ICollection<Address> Addresses { get; set; } = new List<Address>();
    }
}

File: FoodToOrderAngular/src/app/models/restaurant.ts
import { Address } from "./address"
import { Dish } from "./dish"

export class Restaurant{
    id:number
    image:string
    name:string
    dishes:Dish[]
    addresses:Address[]
    ownerid:number

    constructor(i:number,im:string,n:string,add:Address[],d:Dish[],oid:number)
    {
        this.id=i
        this.image=im
        this.name=n
        this.addresses=add
        this.dishes=d
        this.ownerid=oid
    }
}